src/tamp_improv/approaches/improvisational/policies/rl2mpc.py

```diff
@@ -0,0 +1,196 @@
+            check_freq=train_data.config.get("training_record_interval", 100),
+        )
+
+        # Train RL policy
+        print("\nTraining RL component until reward threshold...")
+        self.rl_policy.train(env, train_data, callback=callback)
+
+        # Record if threshold was reached during training
+        # Note: We don't switch to MPC yet - that happens at execution time
+        self._threshold_reached = callback.threshold_reached
+
+    def _initialize_mpc_from_start_state(self) -> None:
+        """Initialize MPC nominal trajectory using RL policy from start
+        state."""
+        if self._env is None or self._start_state is None:
+            raise ValueError("Environment or start state not initialized")
+
+        # Use MPC's wrapped environment for trajectory generation
+        if not hasattr(self.mpc_policy, "env") or self.mpc_policy.env is None:
+            raise ValueError("MPC environment not initialized")
+
+        # Create a separate environment instance for trajectory generation
+        temp_env = deepcopy(self.mpc_policy.env)
+
+        # Generate trajectory from start state using RL
+        current_obs = self._start_state
+        temp_env.reset_from_state(current_obs)  # type: ignore
+        trajectory = []
+
+        for step in range(self.mpc_policy.config.horizon):
+            action = self.rl_policy.get_action(current_obs)
+            trajectory.append(action)
+
+            # Step environment to get next observation
+            next_obs, _, terminated, truncated, _ = temp_env.step(action)
+
+            if terminated or truncated:
+                # If episode ended early, use zero action for remaining steps
+                remaining_steps = self.mpc_policy.config.horizon - (step + 1)
+                zero_action = cast(ActType, np.zeros_like(action))
+                for _ in range(remaining_steps):
+                    trajectory.append(zero_action)
+                break
+
+            current_obs = next_obs
+
+        print(f"Generated MPC trajectory from RL: {trajectory}")
+        trajectory_arr = np.array(trajectory)
+
+        # Set MPC's last solution based on action space type
+        if self.mpc_policy.is_discrete:
+            self.mpc_policy.last_solution = trajectory_arr.astype(np.int32)
+        elif self.mpc_policy.is_multidiscrete:
+            self.mpc_policy.last_solution = trajectory_arr.reshape(
+                -1, self.mpc_policy.action_dims
+            )
+        else:
+            box_space = cast(gym.spaces.Box, self._env.action_space)
+            if box_space.shape:
+                trajectory_arr = trajectory_arr.reshape(-1, *box_space.shape)
+            self.mpc_policy.last_solution = np.clip(
+                trajectory_arr, box_space.low, box_space.high
+            )
+
+        temp_env.close()  # type: ignore[no-untyped-call]
+
+    def get_action(self, obs: ObsType) -> ActType:
+        """Get action using either RL or MPC."""
```

Comment:
for the sake of sanity checking and general interest, can you analyze the number of times that this approach uses RL versus MPC?